client/components/EventsPlanningFilters/EditFilterSchedule.tsx

```diff
@@ -41,88 +49,167 @@ export class EditFilterScheduleComponent extends React.Component<IProps, IState>
+        if (field === 'frequency') {
+            if (value === SCHEDULE_FREQUENCY.WEEKLY || value === SCHEDULE_FREQUENCY.MONTHLY) {
+                schedule.hours = [];
+            } else {
+                delete schedule.hours;
             }
         }
 
         this.setState({schedule});
     }
 
+    updateHour(index: number, value: string) {
+        const hours = [...(this.state.schedule.hours ?? [])];
+
+        hours[index] = value;
+        this.setState({schedule: {...this.state.schedule, hours}});
+    }
+
+    addHour() {
+        const used = new Set(this.state.schedule.hours ?? []);
+        const allHours = Array.from({length: 24}, (_, i) => `${i.toString().padStart(2, '0')}:00`);
+        const next = allHours.find((h) => !used.has(h)) || '00:00';
+
+        const hours = [...(this.state.schedule.hours ?? []), next];
+
+        this.setState({schedule: {...this.state.schedule, hours}});
+    }
+
+    removeHour(index: number) {
+        const hours = [...(this.state.schedule.hours ?? [])];
+
+        hours.splice(index, 1);
+        this.setState({schedule: {...this.state.schedule, hours}});
+    }
+
     getScheduleProfile() {
         const profile = {
             frequency: {enabled: true, index: 1},
             week_days: {enabled: false, index: 2},
             month_day: {enabled: false, index: 3},
-            hour: {enabled: false, index: 4},
         };
 
         switch (this.state.schedule.frequency) {
         case SCHEDULE_FREQUENCY.HOURLY:
             break;
         case SCHEDULE_FREQUENCY.WEEKLY:
             profile.week_days.enabled = true;
-            profile.hour.enabled = true;
             break;
         case SCHEDULE_FREQUENCY.MONTHLY:
             profile.month_day.enabled = true;
-            profile.hour.enabled = true;
             break;
         }
 
         return profile;
     }
 
+    renderTimeInputs() {
+        const {gettext} = superdeskApi.localization;
+        const {schedule} = this.state;
+
+        if (
+            schedule.frequency !== SCHEDULE_FREQUENCY.WEEKLY &&
+            schedule.frequency !== SCHEDULE_FREQUENCY.MONTHLY
+        ) {
+            return null;
+        }
+
+        const hourOptions = Array.from({length: 24}).map((_, i) => {
+            const value = `${i.toString().padStart(2, '0')}:00`;
+
+            return (
+                <option key={value} value={value}>
+                    {value}
+                </option>
+            );
+        });
+
+        return (
+            <div className="form__row">
+                <label className="form__label">{gettext('HOUR')}</label>
+                <div className="form__row-items">
+                    {(schedule.hours ?? []).map((time, idx) => (
+                        <div
+                            className="form__row-item--flex"
+                            key={idx}
+                            style={{marginBottom: '4px', display: 'flex', alignItems: 'center', gap: '0.5rem'}}
```

Comment:
Please do not use static style, it will break the UI with regards to themes.
If need, please request assistance from Fritz and/or Tomas on what to use here instead